doc/How-Tos/DependencyInjection/DependencyInjection-setup.md

```diff
@@ -0,0 +1,88 @@
+# How-To: Use Services with Dependency Injection
+
+Dependency injection (DI) is an important design pattern for building loosely-coupled software that allows for maintainability and testing. This tutorial will walk you through how to register services with a service provider for use throughout your application.
```

Comment:
How do we feel about linking out to other sites for more detailed explanation on DI and design patterns
cc @francoistanguay @jeromelaban

Reply:
This is still a pending question, as there are some helpful resources we could add subtle references to. Notably, these:
https://docs.microsoft.com/en-us/dotnet/core/extensions/dependency-injection
https://docs.microsoft.com/en-us/dotnet/architecture/modern-web-apps-azure/architectural-principles#dependency-inversion

Thoughts @jeromelaban @francoistanguay?